catalyst/loggers/comet.py

```diff
@@ -0,0 +1,264 @@
+                    print("Starting an Offline Experiment")
+                    self.offline_directory = os.getenv('COMET_OFFLINE_DIRECTORY')
+                    self.experiment = comet_ml.OfflineExperiment(
+                        project_name=self.project_name, workspace=self.workspace, offline_directory=self.offline_directory, disabled=self.logging_disabled)
+                else:
+                    self.experiment = comet_ml.Experiment(
+                        project_name=self.project_name, api_key=self.api_key, workspace=self.workspace, disabled=self.logging_disabled
+                    )
+
+            except BaseException as e:
+                experiment = None
+                print(e)
+        else:
+            self.experiment = experiment
+
+        if tags:
+            for tag in tags:
+                self.experiment.add_tag(tag)
+
+    def log_metrics(
+        self,
+        metrics: Dict[str, float],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        key_parameters = [stage_key, loader_key, scope]
+        # Removes all None values from list.
+        passed_key_parmeters = [key_parameter for key_parameter in key_parameters if key_parameter]
+        keys_prefix = None
+        if len(passed_key_parmeters) == 1:
+            keys_prefix = passed_key_parmeters[0]
+        elif len(passed_key_parmeters) > 1:
+            keys_prefix = '_'.join(passed_key_parmeters)
+
+        self.experiment.log_metrics(metrics, step=global_batch_step,
+                                    epoch=global_batch_step, prefix=keys_prefix)
+
+    def log_image(
+        self,
+        tag: str = "images",
+        image: np.ndarray = [],
+        scope: str = None,
+        # experiment info
+        run_key: str = None,
+        global_epoch_step: int = 0,
+        global_batch_step: int = 0,
+        global_sample_step: int = 0,
+        # stage info
+        stage_key: str = None,
+        stage_epoch_len: int = 0,
+        stage_epoch_step: int = 0,
+        stage_batch_step: int = 0,
+        stage_sample_step: int = 0,
+        # loader info
+        loader_key: str = None,
+        loader_batch_len: int = 0,
+        loader_sample_len: int = 0,
+        loader_batch_step: int = 0,
+        loader_sample_step: int = 0,
+    ) -> None:
+        """Logs image to the logger."""
+        self.image_name = tag
+        key_parameters = [stage_key, loader_key, scope]
```

Comment:
Since we will probably do prefix formatting almost every time we log something, I think it might be better to move that logic to it's own internal method. 

e.g. 

`def _format_prefix(prefix_parameters):
    passed_prefix_parameters = [
        parameter for parameter in prefix_parameters if parameter is not None
    ]

    if len(passed_prefix_parameters) > 1:
        prefix = "_".join(passed_prefix_parameters)

    elif len(passed_prefix_parameters) == 1:
        prefix = passed_prefix_parameters[0]

    else:
        prefix = None

    return prefix`